lib/UserInterface/UserInterface.cpp

```diff
@@ -118,6 +121,16 @@ namespace cling {
     std::unique_ptr<TerminalDisplay> D(TerminalDisplay::Create());
     TextInput TI(*R, *D, histfilePath.empty() ? 0 : histfilePath.c_str());
 
+    // Inform Sema about our code complete consumer
+    const CodeCompleteOptions& ClingCodeCompleteOpts = m_MetaProcessor->getInterpreter().getCI()->getFrontendOpts().CodeCompleteOpts;
+    ClingCodeCompleteConsumer* ClingConsumer = new ClingCodeCompleteConsumer(ClingCodeCompleteOpts);
```

Comment:
Who is owning this? Is it ever deleted? If it's meant to be owned by the local scope than please don't use `new` but a local non-pointer variable: `ClingCodeCompleteConsumer ClingConsumer(ClingCodeCompleteOpts);`